### PR TITLE
Allow use of proxy settings

### LINKED ISSUE #29: Request adding support for proxy environments
A lot of companies have limited access from their environments to the internet. Most of them are using web proxies to allow controlled access from systems.

It would help a lot to have such support in your automation.
Setting a proxy would be needed for:
- yum (yum.conf)
- RHSM (rhsm.conf)
- openshift-installer config file 
- bastion server environment variables (in RHEL 8 it is done by the file /etc/profile.d/http_proxy.sh )
- git command

Important is also the no_proxy and that the domain used by the user gets added. Otherwise the installation procedure might get stuck at a point where it shouldn't go through a proxy, but because of the missing no_proxy entry it does.

### PR DESCRIPTION
**Pull request by yussufsh**
Fixes #29

Signed-off-by: Yussuf Shaikh <yussuf@us.ibm.com>

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -78,13 +77,22 @@ Update the following variables specific to OCP.
...
  * `pull_secret_file` : (Optional) Location of the OCP pull-secret file to be used. Default path is 'data/pull-secret.txt'.
  * `dns_forwarders` : (Optional) External DNS servers to forward DNS queries that cannot resolve locally. Eg: `"8.8.8.8; 9.9.9.9"`.
  * `mount_etcd_ramdisk` : (Optional) Flag for mounting etcd directory in the ramdisk. Note that the data will not be persistent.
+ * `rhcos_kernel_options` : (Optional) List of [kernel arguments](https://docs.openshift.com/container-platform/4.4/nodes/nodes/nodes-nodes-working.html#nodes-nodes-kernel-arguments_nodes-nodes-working) for the cluster nodes eg: ["slub_max_order=0","loglevel=7"]. Note that this will be applied after the cluster is installed, hence wait till all the nodes are in `Ready` status before you start using the cluster. Check nodes status using the command `oc get nodes`.
+ * `proxy` : (Optional) Map of below parameters for using a proxy server to setup OCP on a private network.
+    * `server` : Proxy server hostname or IP.
+    * `port` : Proxy port to use (default is 3128).
+    * `user` : Proxy server user for authentication.
+    * `password` : Proxy server password for authentication.
```
Comment (bpradipt) [inline docs/quickstart.md:96]: no_proxy ?

Comment (yussufsh) [reply docs/quickstart.md:96]: It is set automatically to the cluster network CIDR.

I think we don't want to specify additional option for a working OCP cluster setup. If user want to add extra domain would be choice of end users(depending on the workload) and not the cluster admin.

Comment (bpradipt) [reply docs/quickstart.md:96]: ok..

**Review by bpradipt — APPROVED** [review]
lgtm

**Review by Prajyot-Parab — APPROVED** [review]

**Pull request merged**